### DISCUSSION TITLE
Installation instructions for macOS

### DISCUSSION THREAD

**Discussion by cyclaero** [searx/searx#2629] [General]
I wrote an article in German language on my BLog about the possibility of installing und using SearX as a local instance. I gave complete installation instructions for the use of SearX on localhost of macOS and for usage of SearX on a FreeBSD server in the LAN.

The instructions for FreeBSD are meant to be used only until the FreeBSD port becomes updated from currently 0.16.0, which didn’t even work for me, so this is nothing special.

The installation instructions on macOS are different, because here we need to assume that the typical user of macOS does not know how to install python dependencies, edit configuration files and launch a daemon on starting-up the system. I require the user only to download and install the current Python package 3.9.x from python.org, and I provide a shell script which cares for the installation and integration of SearX into the Python installation. In addition I provide pre-configured settings and a launch daemon configuration file, which said script copies to the correct locations. The SearX daemon would then be launched on startup or may be started or stopped by two other scripts which I provide together with the other stuff.

With that, on a vanilla macOS system, SearX is up and running in less than 5 min. I may contribute the macOS installation script and pieces to the SearX repo, if you are interested.

See: https://obsigna.com/articles/1614714756.html#searx_for_macos

In English by courtesy of the Google translator:
https://translate.google.com/translate?hl=&sl=de&tl=en&u=https%3A%2F%2Fobsigna.com%2Farticles%2F1614714756.html%23searx_for_macos

**Comment by jamiejk** [comment]
Hi cyclaero, thanks for this work, I really appreciate it. Could you help with any insight as to why I'd be unable to reach localhost:8000 after getting a 'SearX ready' message in Terminal?

**Reply by jamiejk** [reply]
p.s. I meant 'http://localhost:8888/'!

**Reply by cyclaero** [reply]
Please can you check whether searx is running, e.g. by the following command ...
`ps -axj | grep searx`

... should result in:
`_www              883     1   883      0    0 Ss     ??    0:04.12 /Library/Frameworks/Python.framework/Versions/3.9/Resources/Python.app/Contents/MacOS/Python /Library/Frameworks/Python.framework/Versions/3.9/lib/python3.9/site-packages/searx/webapp.py`

In case it is running check whether disabling the firewall helps.

In case it is not running, execute on the command line:
`bash`

and then:
`SEARX_SETTINGS_PATH=/usr/local/etc/searx-settings.yml /Library/Frameworks/Python.framework/Versions/3.9/Resources/Python.app/Contents/MacOS/Python /Library/Frameworks/Python.framework/Versions/3.9/lib/python3.9/site-packages/searx/webapp.py`

Then report the diagnostic messages.
